src/main/java/org/example/Game2048/Board/BoardGenerator.java

```diff
@@ -0,0 +1,66 @@
+package org.example.Game2048.Board;
+
+import org.example.Game2048.EndOfGameException;
+import org.example.Game2048.Point.Point;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class BoardGenerator {
+    private static final Random RANDOM = new Random();
+
+    public static Board generateNewBoard() {
+        Board board = new Board();
+        Point point1 = new Point(RANDOM.nextInt(4), RANDOM.nextInt(4), 2);
+        Point point2;
+        do {
+            point2 = new Point(RANDOM.nextInt(4), RANDOM.nextInt(4), 2);
+        } while (point1.equals(point2));
+        board.setPointList(List.of(point1, point2));
+        return board;
+    }
+
+    public static Board updateBoard(Board board) {
+        List<Point> points = new ArrayList<>(board.getPointList());
+        Point newPoint;
+        boolean isNewPointEqual;
+        if (points.size() < 16) {
+            do {
+                isNewPointEqual = false;
+                newPoint = new Point(RANDOM.nextInt(4), RANDOM.nextInt(4), 2);
+                for (Point x : points) {
+                    if (newPoint.equals(x)) {
+                        isNewPointEqual = true;
+                        break;
+                    }
+                }
+            } while (isNewPointEqual);
+            points.add(newPoint);
+            board.setPointList(points);
+            return board;
+        } else
+            throw new EndOfGameException();
```

Comment:
EndOfGame to sytuacja nadzwyczajna ? wyjatki to ostatecznosc i powinny byc rzucane tylko w ostatecznosci a to wyglada jak prawidlowe dzialanie :D

Reply:
poza tym rzucajac wyjatek nie wrocisz do glownego menu tylko wywalisz caly program (no chyba ze go tam zlapiesz :P )

Reply:
chcę go później złapać, bo to oznacza koniec gry. Na planszy nie ma więcej ruchów do wykonania